### PR TITLE
[FIX]: Local Development

### PR DESCRIPTION
**Pull request by SujithThirumalaisamy**
Resolves: #42 and #48

- Added local development with services and judge0 in docker
- Added local development with judge0 in docker and databases in anywhere.
- Added Readme for both the setups.
- Made the captcha only work at production

NOTE: Next auth have some issues in the latest release with docker. So local development with next-auth in docker is not possible currently. So web cannot be run in docker. That service is commented in the docker-compose

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,31 +1,30 @@
...
-    ports:
-      - '3000:3000'  # Next.js app
-    depends_on:
-      db:
-        condition: service_healthy
-    volumes:
-      - type: bind
-        source: .
-        target: /usr/src/app
+  # Next Auth Not Working in Docker
```
Comment (siinghd) [inline docker-compose.yml:4]: why does this not work with next auth? any error etc?

File: docker-compose.yml [hunk 2]
```diff
@@ -1,31 +1,30 @@
...
+  #     - '3000:3000'  # Next.js app
+  #   depends_on:
+  #     db:
+  #       condition: service_healthy
+  #   volumes:
+  #     - type: bind
+  #       source: .
+  #       target: /usr/src/app
 
   db:
```
Comment (siinghd) [inline docker-compose.yml:23]: should we add a volume here to persist data?

Comment (SujithThirumalaisamy) [reply docker-compose.yml:23]: This is just for local development. So we need a bind mount

File: docker-compose.yml (hunk 1 above)
Comment (SujithThirumalaisamy) [reply docker-compose.yml:4]: Next auth seems not to be working. When I try to authenticate. The development server stops without any errors

**Pull request merged**